templates/example_visualization.ipynb

```diff
@@ -0,0 +1,263 @@
+    "       'CD206', 'gdTCR', 'iNOS', 'CD68', 'CD36', 'CD8', 'CD3', 'IDO', 'CD11c',\n",
+    "       'CD163', 'CD20', 'CD16', 'IFNg', 'HLA.DR.DQ.DP', 'CD11b', 'CD45',\n",
+    "       'H3K9Ac', 'Keratin.pan', 'CD103', 'MastChyTry', 'MPO', 'NaKATPase',\n",
+    "       'HLA.Class.1', 'Ta', 'Au'], \"PatientID\",algorithm=\"tSNE\", save_dir=\"\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# tSNE Projection of data, with all the Target categories, and Cell Type as the label.\n",
+    "\n",
+    "\n",
+    "dimensionality_reduction.visualize_dimensionality_reduction(all_data,['C', 'Na', 'Si', 'HH3',\n",
+    "       'Vimentin', 'SMA', 'Background', 'Collagen.1', 'Lag3', 'CD4', 'CD14',\n",
+    "       'Foxp3', 'PD.1', 'CD31', 'PD.L1', 'E.cadherin', 'Ki67', 'CD209',\n",
+    "       'CD206', 'gdTCR', 'iNOS', 'CD68', 'CD36', 'CD8', 'CD3', 'IDO', 'CD11c',\n",
+    "       'CD163', 'CD20', 'CD16', 'IFNg', 'HLA.DR.DQ.DP', 'CD11b', 'CD45',\n",
+    "       'H3K9Ac', 'Keratin.pan', 'CD103', 'MastChyTry', 'MPO', 'NaKATPase',\n",
+    "       'HLA.Class.1', 'Ta', 'Au'], \"cell_type\",algorithm=\"tSNE\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# UMAP Projection of data, with all the Target categories, and PatientID as the label. Saves image into current directory.\n",
+    "\n",
+    "\n",
+    "dimensionality_reduction.visualize_dimensionality_reduction(all_data,['C', 'Na', 'Si', 'HH3',\n",
+    "       'Vimentin', 'SMA', 'Background', 'Collagen.1', 'Lag3', 'CD4', 'CD14',\n",
+    "       'Foxp3', 'PD.1', 'CD31', 'PD.L1', 'E.cadherin', 'Ki67', 'CD209',\n",
+    "       'CD206', 'gdTCR', 'iNOS', 'CD68', 'CD36', 'CD8', 'CD3', 'IDO', 'CD11c',\n",
+    "       'CD163', 'CD20', 'CD16', 'IFNg', 'HLA.DR.DQ.DP', 'CD11b', 'CD45',\n",
+    "       'H3K9Ac', 'Keratin.pan', 'CD103', 'MastChyTry', 'MPO', 'NaKATPase',\n",
+    "       'HLA.Class.1', 'Ta', 'Au'], \"PatientID\",algorithm=\"UMAP\", save_dir=\"\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# tSNE Projection of data, with all the Target categories, and Cell Type as the label. \n",
+    "\n",
+    "dimensionality_reduction.visualize_dimensionality_reduction(all_data,['C', 'Na', 'Si', 'HH3',\n",
+    "       'Vimentin', 'SMA', 'Background', 'Collagen.1', 'Lag3', 'CD4', 'CD14',\n",
+    "       'Foxp3', 'PD.1', 'CD31', 'PD.L1', 'E.cadherin', 'Ki67', 'CD209',\n",
+    "       'CD206', 'gdTCR', 'iNOS', 'CD68', 'CD36', 'CD8', 'CD3', 'IDO', 'CD11c',\n",
+    "       'CD163', 'CD20', 'CD16', 'IFNg', 'HLA.DR.DQ.DP', 'CD11b', 'CD45',\n",
+    "       'H3K9Ac', 'Keratin.pan', 'CD103', 'MastChyTry', 'MPO', 'NaKATPase',\n",
+    "       'HLA.Class.1', 'Ta', 'Au'], \"cell_type\",algorithm=\"UMAP\", save_dir=\"\")"
```

Comment:
Please assign `['C', 'Na', ...]` to a variable rather than repeating the same list for every call to visualize_dimensionality_reduction 😅.